native/src/routes/CategoriesContainer.tsx

```diff
@@ -56,11 +57,22 @@ const CategoriesContainer = ({ navigation, route }: CategoriesContainerProps): R
   )
   const previousLanguageCode = usePreviousProp({ prop: languageCode, onPropChange: onLanguageChange })
 
+  const embeddedOffers = useEmbeddedOffers({
+    category,
+    cityCode,
+    languageCode,
+  })
+
   const error =
     data?.categories && !category && previousLanguageCode === languageCode ? ErrorCode.PageNotFound : response.error
 
+  const refresh = useCallback(() => {
+    response.refresh()
+    embeddedOffers.refresh()
+  }, [response, embeddedOffers])
+
   return (
-    <LoadingErrorHandler {...response} error={error} scrollView>
+    <LoadingErrorHandler refresh={refresh} loading={response.loading} error={error} scrollView>
```

Comment:
I changed this part again, so that the page content is already displayed, while the offer might still be loading. In my opinion this provides a much better user experience.